File: Using TDD with ASP.NET MVC/[VB]-Using TDD with ASP.NET MVC/VB/MvcContacts/Models/ContactMD.vb
' <snippet1>
Imports System.ComponentModel.DataAnnotations

Public Class ContactMD

    <ScaffoldColumn(False)>
    Public Property Id() As Object
    <Required()>
    Public Property FirstName() As Object
    <Required()>
    Public Property LastName() As Object
    <RegularExpression("^\d{3}-?\d{3}-?\d{4}$")>
    Public Property Phone() As Object

    <Required()>
    <DataType(DataType.EmailAddress)>
    <RegularExpression("^([0-9a-zA-Z]([-.\w]*[0-9a-zA-Z])*@([0-9a-zA-Z][-\w]*[0-9a-zA-Z]\.)+[a-zA-Z]{2,9})$")>
    Public Property Email() As Object

End Class

' </snippet1>

File: Using TDD with ASP.NET MVC/[C#]-Using TDD with ASP.NET MVC/C#/MvcContacts/Models/ContactMD.cs
// <snippet1>
using System.ComponentModel.DataAnnotations;

namespace MvcContacts.Models {
    [MetadataType(typeof(ContactMD))]
    public partial class Contact {
        public class ContactMD {
            [ScaffoldColumn(false)]
            public object Id { get; set; }
            [Required()]
            public object FirstName { get; set; }
            [Required()]
            public object LastName { get; set; }
            [RegularExpression(@"^\d{3}-?\d{3}-?\d{4}$")]
            public object Phone { get; set; }
            [Required()]
            [DataType(DataType.EmailAddress)]
            [RegularExpression(@"^([0-9a-zA-Z]([-.\w]*[0-9a-zA-Z])*@([0-9a-zA-Z][-\w]*[0-9a-zA-Z]\.)+[a-zA-Z]{2,9})$")]
            public object Email { get; set; }
        }
    }
} 
// </snippet1>

